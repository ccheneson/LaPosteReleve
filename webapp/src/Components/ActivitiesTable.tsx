import React, { useState, useRef } from "react";
import ActivityDetails, { ActivityDetailsMonthly } from "./ActivityDetails"
import index from "../index.css";
import Search from "./Search";

type CssByDate = {
    previousDate: string,
    style: string
}


const monthByIndex = [
    "January", "February", "March", "April", "May", "June",
    "July", "August", "September", "October", "November", "December",
]

type ActivitiesPerMonthJson = {
    month_index: number,
    stats: {
        amount_plus: number,
        amount_minus: number
    },
    activities: ActivityDetailsJson[]
}

type ActivityDetailsJson = {
    row_id: number,
    date: string,
    statement: string,
    amount: number,
    tag_pattern_id: number | null
};

type BalanceJson = {
    date: string,
    amount: number
};

type TagsJson = {
    [tag_pattern_id: number]: string[]
};

type ActivitiesTableProps = {
    activities: ActivitiesPerMonthJson[],
    balance: BalanceJson
    tags: TagsJson
}

const ActivitiesTable = (props: ActivitiesTableProps) => {

    let balance = props.balance;
    let tags = props.tags;

    const [activities, setActivities] = useState<ActivitiesPerMonthJson[]>(props.activities)
    const activitiesJson = useRef(props.activities);

    var cssByDate: CssByDate = { previousDate: "", style: "" };

    const toggleRowStyle = (date: string) => {
        if (cssByDate === undefined || date !== cssByDate.previousDate) {
            cssByDate.style = cssByDate.style === index.colorRowG ? index.colorRowB : index.colorRowG;
            cssByDate.previousDate = date;
        }
        return cssByDate.style;
    }

    const formatDate = (d: string) => {
        const date = new Date(d)
        return date.getDate().toString().padStart(2, '0') + "/" + (date.getMonth() + 1).toString().padStart(2, '0') + "/" + date.getFullYear()
    }

    const SEARCH_UNTAGGED = "null";
    const check_null_search = (pattern: string) => {
        if (pattern.length <= SEARCH_UNTAGGED.length && SEARCH_UNTAGGED.startsWith(pattern)) {
            return true;
        } else {
            return false;
        }
    }

    const updateSearchResult = (activitiesJson: ActivitiesPerMonthJson[]) => (e: React.FormEvent<HTMLInputElement>) => {
        const searchPattern = e.currentTarget.value.toLowerCase();
        const checkPatternInTags = (tagPatternId: number | null, searchPattern: string) => {
            if (tagPatternId == null && check_null_search(searchPattern)) {
                return true;
            } else if (tagPatternId == null) {
                return false;
            } else if (tags && tagPatternId in tags) {
                return tags[tagPatternId].filter(e => e.toString().toLowerCase().includes(searchPattern)).length > 0;
            } else {
                return false;
            }
        }
        const updated = activitiesJson.map(e => {
            const newActivities = e.activities
                .filter(
                    a => a.amount.toString().toLowerCase().includes(searchPattern) ||
                        a.date.toString().toLowerCase().includes(searchPattern) ||
                        a.statement.toLowerCase().includes(searchPattern) ||
                        checkPatternInTags(a.tag_pattern_id, searchPattern)
                )

            return { month_index: e.month_index, stats: e.stats, activities: newActivities }
        })
        setActivities(updated);
    }

    const checkIfResultEmpty = (activities: ActivitiesPerMonthJson[]) => {
        return activities.filter((e: ActivitiesPerMonthJson) => e.activities.length > 0).length == 0;
    }

    return (
        <div>
            {activities.length === 0
                ?
                <div>
                    <h3>Error while loading page: can not reach data source</h3>
                </div>
                :
                <div>
                    <Search onChange={updateSearchResult(activitiesJson.current)} />
                    <br />
                    {checkIfResultEmpty(activities)
                        ?
                        <div>
                            <h3>No result matching</h3>
                        </div>
                        :
                        <table id="account" className={index.large_table}>
                            <tbody>
                                <tr>
                                    <th>Month</th>
                                    <th>Date</th>
                                    <th>Libelle</th>
                                    <th>Tags</th>
                                    <th>Montant {balance ? "au " + formatDate(balance?.date) : ""}<br />
                                        {`${balance?.amount} â‚¬`}
                                    </th>
                                </tr>
                                {
                                    activities.map((activitiesPerMonth: ActivitiesPerMonthJson) =>
                                        activitiesPerMonth.activities.map((activity, i) =>
                                            (i === 0) ?
                                                <ActivityDetailsMonthly key={activity.row_id} className={toggleRowStyle(activity.date)}
                                                    date={formatDate(activity.date)} statement={activity.statement}
                                                    amount={activity.amount}
                                                    statsPlus={`+${activitiesPerMonth.stats.amount_plus.toFixed(2)}`}
                                                    statsMinus={`${activitiesPerMonth.stats.amount_minus.toFixed(2)}`}
                                                    month={monthByIndex[activitiesPerMonth.month_index - 1]}
                                                    tags={tags && activity.tag_pattern_id && activity.tag_pattern_id in tags ? tags[activity.tag_pattern_id] : []} />
                                                :
                                                <ActivityDetails key={activity.row_id} className={toggleRowStyle(activity.date)}
                                                    date={formatDate(activity.date)} statement={activity.statement}
                                                    amount={activity.amount} tags={tags && activity.tag_pattern_id && activity.tag_pattern_id in tags ? tags[activity.tag_pattern_id] : []} />
                                        )
                                    )}
                            </tbody>
                        </table>
                    }
                </div>
            }
        </div >
    );
};

export default ActivitiesTable;
export { monthByIndex, ActivitiesPerMonthJson, ActivityDetailsJson };
