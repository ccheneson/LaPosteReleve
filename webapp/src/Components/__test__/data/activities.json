[
    {
      "month_index":7,
      "stats":{
        "amount_plus":0.0,
        "amount_minus":-73.0
      },
      "activities":[
        {
          "row_id":8,
          "date":"2021-07-26",
          "statement":"BUY SOMETHING FREE MOBILE 01",
          "amount":-30.0,
          "tag_pattern_id":5
        },
        {
          "row_id":6,
          "date":"2021-07-16",
          "statement":"BUY SOMETHING 02",
          "amount":-21.0,
          "tag_pattern_id":null
        },
        {
          "row_id":2,
          "date":"2021-07-12",
          "statement":"BUY SOMETHING 03",
          "amount":-22.0,
          "tag_pattern_id":null
        }
      ]
    },
    {
      "month_index":6,
      "stats":{
        "amount_plus":0.0,
        "amount_minus":-122.0
      },
      "activities":[
        {
          "row_id":4,
          "date":"2021-06-09",
          "statement":"BUY SOMETHING EDF 04",
          "amount":-22.0,
          "tag_pattern_id":1
        },
        {
          "row_id":1,
          "date":"2021-06-09",
          "statement":"BUY SOMETHING 04",
          "amount":-64.0,
          "tag_pattern_id":null
        },
        {
          "row_id":5,
          "date":"2021-06-09",
          "statement":"BUY RETRAIT 04",
          "amount":-36.0,
          "tag_pattern_id":3
        }
      ]
    },
    {
      "month_index":5,
      "stats":{
        "amount_plus":0.0,
        "amount_minus":-216.0
      },
      "activities":[
        {
          "row_id":16,
          "date":"2021-05-26",
          "statement":"BUY SOMETHING 01",
          "amount":-54.0,
          "tag_pattern_id":null
        },
        {
          "row_id":14,
          "date":"2021-05-12",
          "statement":"BUY SOMETHING 03",
          "amount":-20.0,
          "tag_pattern_id":null
        },
        {
          "row_id":11,
          "date":"2021-05-10",
          "statement":"BUY SOMETHING 02",
          "amount":-45.0,
          "tag_pattern_id":null
        },
        {
          "row_id":18,
          "date":"2021-05-09",
          "statement":"BUY SOMETHING VIREMENT",
          "amount":-10.0,
          "tag_pattern_id":2
        },
        {
          "row_id":3,
          "date":"2021-05-09",
          "statement":"BUY SOMETHING 07",
          "amount":-44.0,
          "tag_pattern_id":null
        },
        {
          "row_id":9,
          "date":"2021-05-09",
          "statement":"BUY RETRAIT 05",
          "amount":-11.0,
          "tag_pattern_id":3
        },
        {
          "row_id":7,
          "date":"2021-05-08",
          "statement":"BUY SOMETHING 06",
          "amount":-32.0,
          "tag_pattern_id":null
        }
      ]
    },
    {
      "month_index":4,
      "stats":{
        "amount_plus":0.0,
        "amount_minus":-367.0
      },
      "activities":[
        {
          "row_id":12,
          "date":"2021-04-09",
          "statement":"BUY SOMETHING LOYER 05",
          "amount":-300.0,
          "tag_pattern_id":4
        },
        {
          "row_id":17,
          "date":"2021-04-09",
          "statement":"BUY SOMETHING EDF 04",
          "amount":-11.0,
          "tag_pattern_id":1
        },
        {
          "row_id":15,
          "date":"2021-04-09",
          "statement":"BUY SOMETHING 04",
          "amount":-56.0,
          "tag_pattern_id":null
        }
      ]
    },
    {
      "month_index":3,
      "stats":{
        "amount_plus":0.0,
        "amount_minus":-68.0
      },
      "activities":[
        {
          "row_id":21,
          "date":"2021-03-26",
          "statement":"BUY SOMETHING FREE MOBILE 01",
          "amount":-20.0,
          "tag_pattern_id":5
        },
        {
          "row_id":22,
          "date":"2021-03-16",
          "statement":"BUY SOMETHING 02",
          "amount":-10.0,
          "tag_pattern_id":null
        },
        {
          "row_id":24,
          "date":"2021-03-12",
          "statement":"BUY SOMETHING 03",
          "amount":-15.0,
          "tag_pattern_id":null
        },
        {
          "row_id":13,
          "date":"2021-03-09",
          "statement":"BUY SOMETHING 07",
          "amount":-5.0,
          "tag_pattern_id":null
        },
        {
          "row_id":25,
          "date":"2021-03-09",
          "statement":"BUY SOMETHING 04",
          "amount":-8.0,
          "tag_pattern_id":null
        },
        {
          "row_id":10,
          "date":"2021-03-08",
          "statement":"BUY SOMETHING 06",
          "amount":-10.0,
          "tag_pattern_id":null
        }
      ]
    },
    {
      "month_index":2,
      "stats":{
        "amount_plus":0.0,
        "amount_minus":-489.53
      },
      "activities":[
        {
          "row_id":19,
          "date":"2021-02-09",
          "statement":"BUY SOMETHING LOYER 05",
          "amount":-300.0,
          "tag_pattern_id":4
        },
        {
          "row_id":20,
          "date":"2021-02-09",
          "statement":"BUY SOMETHING EDF 04",
          "amount":-41.0,
          "tag_pattern_id":1
        },
        {
          "row_id":23,
          "date":"2021-02-09",
          "statement":"BUY SOMETHING 07",
          "amount":-52.53,
          "tag_pattern_id":null
        },
        {
          "row_id":26,
          "date":"2021-02-08",
          "statement":"BUY SOMETHING 06",
          "amount":-44.0,
          "tag_pattern_id":null
        },
        {
          "row_id":27,
          "date":"2021-02-07",
          "statement":"BUY SOMETHING 04",
          "amount":-52.0,
          "tag_pattern_id":null
        }
      ]
    }
  ]