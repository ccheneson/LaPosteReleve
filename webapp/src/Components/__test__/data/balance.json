{"date":"2021-07-28","amount":132.85}