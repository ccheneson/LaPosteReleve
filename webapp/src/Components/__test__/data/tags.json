{
    "1":[
      "EDF"
    ],
    "5":[
      "FREEMOBILE",
      "PARIS"
    ],
    "4":[
      "LOYER",
      "PARIS"
    ],
    "3":[
      "RETRAIT"
    ],
    "2":[
      "VIREMENT_BANCAIRE"
    ]
  }