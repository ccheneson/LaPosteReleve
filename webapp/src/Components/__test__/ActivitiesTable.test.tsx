import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { ReactNode } from 'react';
import ActivitiesTable from "../ActivitiesTable";
import activities from "./data/activities.json";
import balance from "./data/balance.json";
import tags from "./data/tags.json";

describe('ActivitiesTable', () => {
  let wrapper: ReactNode = render(
    <ActivitiesTable activities={activities} balance={balance} tags={tags} />
  );

  it('should match the snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  });


  afterEach(cleanup);
})