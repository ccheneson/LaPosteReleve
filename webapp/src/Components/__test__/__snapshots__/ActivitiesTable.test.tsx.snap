// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivitiesTable should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            Search: 
            <input
              type="text"
            />
             ("null" for untagged activities)
          </div>
          <br />
          <table
            class="large_table"
            id="account"
          >
            <tbody>
              <tr>
                <th>
                  Month
                </th>
                <th>
                  Date
                </th>
                <th>
                  Libelle
                </th>
                <th>
                  Tags
                </th>
                <th>
                  Montant 
                  au 28/07/2021
                  <br />
                  132.85 â‚¬
                </th>
              </tr>
              <tr
                class="colorRowG"
              >
                <td
                  class="month"
                >
                  July
                  <br />
                  +0.00
                  <br />
                  -73.00
                </td>
                <td>
                  26/07/2021
                </td>
                <td>
                  BUY SOMETHING FREE MOBILE 01
                </td>
                <td>
                  FREEMOBILE, PARIS
                </td>
                <td
                  class="amountMinus"
                >
                  -30.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  16/07/2021
                </td>
                <td>
                  BUY SOMETHING 02
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -21.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td />
                <td>
                  12/07/2021
                </td>
                <td>
                  BUY SOMETHING 03
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -22.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td
                  class="month"
                >
                  June
                  <br />
                  +0.00
                  <br />
                  -122.00
                </td>
                <td>
                  09/06/2021
                </td>
                <td>
                  BUY SOMETHING EDF 04
                </td>
                <td>
                  EDF
                </td>
                <td
                  class="amountMinus"
                >
                  -22.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/06/2021
                </td>
                <td>
                  BUY SOMETHING 04
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -64.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/06/2021
                </td>
                <td>
                  BUY RETRAIT 04
                </td>
                <td>
                  RETRAIT
                </td>
                <td
                  class="amountMinus"
                >
                  -36.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td
                  class="month"
                >
                  May
                  <br />
                  +0.00
                  <br />
                  -216.00
                </td>
                <td>
                  26/05/2021
                </td>
                <td>
                  BUY SOMETHING 01
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -54.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  12/05/2021
                </td>
                <td>
                  BUY SOMETHING 03
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -20.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td />
                <td>
                  10/05/2021
                </td>
                <td>
                  BUY SOMETHING 02
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -45.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/05/2021
                </td>
                <td>
                  BUY SOMETHING VIREMENT
                </td>
                <td>
                  VIREMENT_BANCAIRE
                </td>
                <td
                  class="amountMinus"
                >
                  -10.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/05/2021
                </td>
                <td>
                  BUY SOMETHING 07
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -44.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/05/2021
                </td>
                <td>
                  BUY RETRAIT 05
                </td>
                <td>
                  RETRAIT
                </td>
                <td
                  class="amountMinus"
                >
                  -11.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td />
                <td>
                  08/05/2021
                </td>
                <td>
                  BUY SOMETHING 06
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -32.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td
                  class="month"
                >
                  April
                  <br />
                  +0.00
                  <br />
                  -367.00
                </td>
                <td>
                  09/04/2021
                </td>
                <td>
                  BUY SOMETHING LOYER 05
                </td>
                <td>
                  LOYER, PARIS
                </td>
                <td
                  class="amountMinus"
                >
                  -300.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/04/2021
                </td>
                <td>
                  BUY SOMETHING EDF 04
                </td>
                <td>
                  EDF
                </td>
                <td
                  class="amountMinus"
                >
                  -11.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/04/2021
                </td>
                <td>
                  BUY SOMETHING 04
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -56.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td
                  class="month"
                >
                  March
                  <br />
                  +0.00
                  <br />
                  -68.00
                </td>
                <td>
                  26/03/2021
                </td>
                <td>
                  BUY SOMETHING FREE MOBILE 01
                </td>
                <td>
                  FREEMOBILE, PARIS
                </td>
                <td
                  class="amountMinus"
                >
                  -20.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  16/03/2021
                </td>
                <td>
                  BUY SOMETHING 02
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -10.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td />
                <td>
                  12/03/2021
                </td>
                <td>
                  BUY SOMETHING 03
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -15.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/03/2021
                </td>
                <td>
                  BUY SOMETHING 07
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -5.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/03/2021
                </td>
                <td>
                  BUY SOMETHING 04
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -8.00
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td />
                <td>
                  08/03/2021
                </td>
                <td>
                  BUY SOMETHING 06
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -10.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td
                  class="month"
                >
                  February
                  <br />
                  +0.00
                  <br />
                  -489.53
                </td>
                <td>
                  09/02/2021
                </td>
                <td>
                  BUY SOMETHING LOYER 05
                </td>
                <td>
                  LOYER, PARIS
                </td>
                <td
                  class="amountMinus"
                >
                  -300.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/02/2021
                </td>
                <td>
                  BUY SOMETHING EDF 04
                </td>
                <td>
                  EDF
                </td>
                <td
                  class="amountMinus"
                >
                  -41.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  09/02/2021
                </td>
                <td>
                  BUY SOMETHING 07
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -52.53
                </td>
              </tr>
              <tr
                class="colorRowG"
              >
                <td />
                <td>
                  08/02/2021
                </td>
                <td>
                  BUY SOMETHING 06
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -44.00
                </td>
              </tr>
              <tr
                class="colorRowB"
              >
                <td />
                <td>
                  07/02/2021
                </td>
                <td>
                  BUY SOMETHING 04
                </td>
                <td />
                <td
                  class="amountMinus"
                >
                  -52.00
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          Search: 
          <input
            type="text"
          />
           ("null" for untagged activities)
        </div>
        <br />
        <table
          class="large_table"
          id="account"
        >
          <tbody>
            <tr>
              <th>
                Month
              </th>
              <th>
                Date
              </th>
              <th>
                Libelle
              </th>
              <th>
                Tags
              </th>
              <th>
                Montant 
                au 28/07/2021
                <br />
                132.85 â‚¬
              </th>
            </tr>
            <tr
              class="colorRowG"
            >
              <td
                class="month"
              >
                July
                <br />
                +0.00
                <br />
                -73.00
              </td>
              <td>
                26/07/2021
              </td>
              <td>
                BUY SOMETHING FREE MOBILE 01
              </td>
              <td>
                FREEMOBILE, PARIS
              </td>
              <td
                class="amountMinus"
              >
                -30.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                16/07/2021
              </td>
              <td>
                BUY SOMETHING 02
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -21.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td />
              <td>
                12/07/2021
              </td>
              <td>
                BUY SOMETHING 03
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -22.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td
                class="month"
              >
                June
                <br />
                +0.00
                <br />
                -122.00
              </td>
              <td>
                09/06/2021
              </td>
              <td>
                BUY SOMETHING EDF 04
              </td>
              <td>
                EDF
              </td>
              <td
                class="amountMinus"
              >
                -22.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/06/2021
              </td>
              <td>
                BUY SOMETHING 04
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -64.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/06/2021
              </td>
              <td>
                BUY RETRAIT 04
              </td>
              <td>
                RETRAIT
              </td>
              <td
                class="amountMinus"
              >
                -36.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td
                class="month"
              >
                May
                <br />
                +0.00
                <br />
                -216.00
              </td>
              <td>
                26/05/2021
              </td>
              <td>
                BUY SOMETHING 01
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -54.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                12/05/2021
              </td>
              <td>
                BUY SOMETHING 03
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -20.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td />
              <td>
                10/05/2021
              </td>
              <td>
                BUY SOMETHING 02
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -45.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/05/2021
              </td>
              <td>
                BUY SOMETHING VIREMENT
              </td>
              <td>
                VIREMENT_BANCAIRE
              </td>
              <td
                class="amountMinus"
              >
                -10.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/05/2021
              </td>
              <td>
                BUY SOMETHING 07
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -44.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/05/2021
              </td>
              <td>
                BUY RETRAIT 05
              </td>
              <td>
                RETRAIT
              </td>
              <td
                class="amountMinus"
              >
                -11.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td />
              <td>
                08/05/2021
              </td>
              <td>
                BUY SOMETHING 06
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -32.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td
                class="month"
              >
                April
                <br />
                +0.00
                <br />
                -367.00
              </td>
              <td>
                09/04/2021
              </td>
              <td>
                BUY SOMETHING LOYER 05
              </td>
              <td>
                LOYER, PARIS
              </td>
              <td
                class="amountMinus"
              >
                -300.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/04/2021
              </td>
              <td>
                BUY SOMETHING EDF 04
              </td>
              <td>
                EDF
              </td>
              <td
                class="amountMinus"
              >
                -11.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/04/2021
              </td>
              <td>
                BUY SOMETHING 04
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -56.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td
                class="month"
              >
                March
                <br />
                +0.00
                <br />
                -68.00
              </td>
              <td>
                26/03/2021
              </td>
              <td>
                BUY SOMETHING FREE MOBILE 01
              </td>
              <td>
                FREEMOBILE, PARIS
              </td>
              <td
                class="amountMinus"
              >
                -20.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                16/03/2021
              </td>
              <td>
                BUY SOMETHING 02
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -10.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td />
              <td>
                12/03/2021
              </td>
              <td>
                BUY SOMETHING 03
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -15.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/03/2021
              </td>
              <td>
                BUY SOMETHING 07
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -5.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/03/2021
              </td>
              <td>
                BUY SOMETHING 04
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -8.00
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td />
              <td>
                08/03/2021
              </td>
              <td>
                BUY SOMETHING 06
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -10.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td
                class="month"
              >
                February
                <br />
                +0.00
                <br />
                -489.53
              </td>
              <td>
                09/02/2021
              </td>
              <td>
                BUY SOMETHING LOYER 05
              </td>
              <td>
                LOYER, PARIS
              </td>
              <td
                class="amountMinus"
              >
                -300.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/02/2021
              </td>
              <td>
                BUY SOMETHING EDF 04
              </td>
              <td>
                EDF
              </td>
              <td
                class="amountMinus"
              >
                -41.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                09/02/2021
              </td>
              <td>
                BUY SOMETHING 07
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -52.53
              </td>
            </tr>
            <tr
              class="colorRowG"
            >
              <td />
              <td>
                08/02/2021
              </td>
              <td>
                BUY SOMETHING 06
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -44.00
              </td>
            </tr>
            <tr
              class="colorRowB"
            >
              <td />
              <td>
                07/02/2021
              </td>
              <td>
                BUY SOMETHING 04
              </td>
              <td />
              <td
                class="amountMinus"
              >
                -52.00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
